src/Riak/Command/Search/Response.php

```diff
@@ -35,7 +35,7 @@ public function __construct($statusCode, $headers = [], $body = '')
         parent::__construct($statusCode, $headers, $body);
 
         // make sure body is not only whitespace
-        if (trim($body)) {
+        if (trim($body) && $this->statusCode != 400) {
```

Comment:
^^^^ what @javajolt said, easier to test for success here, I think. 200, 201, 204, maybe?

I have this reference bookmarked: http://www.restapitutorial.com/httpstatuscodes.html